core/src/machine/mod.rs

```diff
@@ -88,6 +91,12 @@ impl Machine {
     pub fn store_hl(&mut self, byte: Byte) {
         self.store_byte(self.cpu.hl(), byte);
     }
+
+    /// Pushes the given word onto the stack.
+    pub fn push(&mut self, word: Word) {
+        self.cpu.sp -= 2u16;
+        self.store_word(self.cpu.sp, word);
+    }
```

Comment:
> Mhhhh, I think I would actually move this method to step.rs, because it's only used there (as a helper method), right?

I don't think this a good idea. All methods should be in the same place, no matter where they are currently used. In my opinion `step.rs` with the `step` method is an exception, because it has so many lines of code.